File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/http/common/HttpInteractionHandler.java
package io.lsdconsulting.interceptors.http.common;

import java.util.Map;

public interface HttpInteractionHandler {
    void handleRequest(String method, Map<String, String> requestHeaders, String path, String body);

    void handleResponse(String statusMessage, Map<String, String> requestHeaders, String path, String body);
}

File: lsd-interceptors/src/main/kotlin/io/lsdconsulting/interceptors/http/common/HttpInteractionHandler.kt
package io.lsdconsulting.interceptors.http.common

import java.time.Duration

interface HttpInteractionHandler {
    fun handleRequest(method: String, requestHeaders: Map<String, String>, path: String, body: String)
    fun handleResponse(
        statusMessage: String,
        requestHeaders: Map<String, String>,
        responseHeaders: Map<String, String>,
        path: String,
        body: String,
        duration: Duration
    )
}
